tests/unit/parsers/test.manifestjson.js

```diff
@@ -172,6 +172,21 @@ describe('ManifestJSONParser', () => {
       const metadata = manifestJSONParser.getMetadata();
       expect(metadata.id).toEqual(null);
     });
+
+    it('should fail if "homepage_url" links to AMO domain names', () => {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON({
+        homepage_url: 'http://addons.mozilla.org',
+      });
+      const manifestJSONParser = new ManifestJSONParser(
+        json,
+        addonLinter.collector
+      );
+      expect(manifestJSONParser.isValid).toEqual(false);
+      const { errors } = addonLinter.collector;
+      expect(errors.length).toEqual(1);
+      expect(errors[0].code).toEqual('RESTRICTED_HOMEPAGE_URL');
+    });
```

Comment:
Let's also test the opposite as I did mention in another comment (if the homepage url is not forbidden, then there shouldn't be an error and the manifest should be valid).